integration/integration.py

```diff
@@ -173,34 +186,72 @@ def try_wait(process, timeout):
             LOGGER.debug('standard error:')
             for line in open(stderr).readlines()[-100:]:
                 LOGGER.debug(f'    {line}')
+            return result
         # Perform baseline update or comparison.
         baseline = baseline_dir / f'{step_id}.ref'
+        output_format = deduce_format(step.command)
```

Comment:
Couldn't you just try to attempt to parse the output as JSON with builtin Python tools?

Reply:
What would be the advantage of that approach?

Reply:
It seemed to be simpler and more maintainable than looking at the command structure. This was just an idea, though, no need to budge.